language_names/build_db.py

```diff
@@ -0,0 +1,38 @@
+import csv
+import sqlite3
+import textwrap
+
+
+def insert_values(c, filename, tablename):
+    c.execute(textwrap.dedent("""
+        CREATE TABLE IF NOT EXISTS {} (
+            id INTEGER PRIMARY KEY,
+            lg TEXT,
+            inLg TEXT,
+            name TEXT,
+            UNIQUE(lg, inLg) ON CONFLICT IGNORE);
+        """.format(tablename)))
+    with open(filename, 'r') as f:
+        reader = csv.DictReader(f, delimiter='\t')
+        for row in reader:
+            c.execute('INSERT INTO {} VALUES (?, ?, ?, ?)'.format(tablename), (None, row['lg'], row['inLg'], row['name']))
+
+
+def populate_database():
+    conn = sqlite3.connect('langNames.db')
+    c = conn.cursor()
+    c.execute('PRAGMA foreign_keys=OFF;')
+    c.execute('BEGIN TRANSACTION;')
+    insert_values(c, 'language_names/fixes.tsv', 'fixes')
+    insert_values(c, 'language_names/additions.tsv', 'additions')
+    insert_values(c, 'language_names/turkic_fixes.tsv', 'fixes')
+    insert_values(c, 'language_names/turkic_langNames.tsv', 'languageNames')
+    insert_values(c, 'language_names/scraped.tsv', 'languageNames')
+    insert_values(c, 'language_names/scraped-sil.tsv', 'languageNames')
+    insert_values(c, 'language_names/variants.tsv', 'languageNames')
```

Comment:
This doesn't actually result in everything ending up in the `languageNames` table. Note the final lines in `manual.sql`:

```
INSERT INTO "languageNames" (lg, inLg, name) select lg, inLg, name from "fixes";
INSERT INTO "languageNames" (lg, inLg, name) select lg, inLg, name from "additions";
```

Honestly, I don't think you even need the other tables. But you do need to use ON CONFLICT REPLACE with the scraped versions being inserted first.

Reply:
I checked for the original database and found it contains 3 dbs but now I will eliminate the other two dbs if they aren't needed.

Reply:
The other two don't actually get used. Per my paste, their data just gets copied into the first table and APy only queries it. Feel free to verify this.

Reply:
The only difference I found in these files is there are around 50 lines in turkic.sql that goes only to fixes. I don't know if they are of any use. Should I pass them in languageNames too. Doing this will result in only one database in the langNames.db file i.e., languageNames.